source/main.c

```diff
@@ -17,29 +19,63 @@ int main(void) {
 	Sleep(100);
 
 	Boot_Log("Hello World!");
+	FILE* fp;
+	char str[16];
+	char* filename = "c:\\input.txt";
+
+	fp = fopen(filename, "r");
+	if (fp == NULL) {
+		printf("Could not open file %s", filename);
+		/*return 1;*/
+	}
+	while (fgets(str, 16, fp) != NULL) {
```

Comment:
You are overwriting the previous value here. Let us suppose your file has more than 16 characters, we will have the last 16 characters at the end. 

If your file has only 16 bytes and if you want to encrypt only 16 bytes. that's fine, but if you would like to handle a file greater than 16 bytes (as a C training), you might get the file size first (google please), create a dynamic memory using malloc, and use a pointer to increase the offset to not overwrite the previous value.